modules/core/src/copy.cpp

```diff
@@ -628,6 +628,45 @@ Mat& Mat::setTo(InputArray _value, InputArray _mask)
 
     CV_IPP_RUN_FAST(ipp_Mat_setTo_Mat(*this, value, mask), *this)
 
+    const bool checkMemsetUsability = mask.empty();
+    if (checkMemsetUsability)
```

Comment:
As I can see, there is `can_apply_memset` method, can it be reused or extended?

Reply:
Indeed, you are right, and we can go even further.
with an empty mask, why is `Mat::setTo()` not delegated to `Mat::operator=(const Scalar&)` ?

Reply:
I think it make sense to do such experiment. However it won't be a simple shortcut from `setTo` to the operator due there is no option to get a `Scalar` from `InputArray`.

Reply:
Now we have two diffferent implementations trying to use `memset()`
The new one has two advantages :

1. it tries to avoid overhead for simple continuous and 2D cases
2. it detects more cases of "uniform byte compatible with a memset()" when unrolling the scalar (at the cost of a little more complexity)

Both 1. and 2. might or might not worth the new code.

So we have 2*2 options :

- keep old implementation, or migrate to new implementation
- rewrite `operator=(const cv::Scalar& s)` to be `setTo(s, cv::noArray())`, or make `setTo(m, empty)` call `operator=(const cv::Scalar& s)`

How would we decide what's best ?